data_ocean/utils.py

```diff
@@ -0,0 +1,44 @@
+"""
+Here we are collecting alf functions for cleaning xml records from whitespaces and /
+abbreviations, splitting tag.texts and converting strings with date info to datetimes
+"""
+import datetime
+import re
+
+
+# changing to lowercase, deleting 'р.', 'м.', 'с.', 'смт.', 'смт', 'сщ.', 'с/рада.', 'сщ/рада.', /
+# 'вул.' from a string
+def clean_name(name):
+    name = name.lower()
+    return re.sub(r'р\.|м\.|с\.|смт|смт\.|сщ\.|с/рада\.|сщ/рада\.', "", name).trim()
+
+# changing to lowercase and abbreviations to full forms like 'р-н' to 'район' /
+# (see all cases in short_to_full_dict)
+def change_to_full_name(name):
+    #a dictionary for storing all abbreviations and full forms
+    short_to_full_dict = {
+        'р-н': 'район',
+        'вул.': 'вулиця ',
+        'бульв.': 'бульвар ',
+        'просп.': 'проспект ',
+        'пров.': 'провулок ',
+        'пл.': 'площа ',
+        'обл.': 'область',
+        'области': 'області'
+    }
+    name = name.lower()
+    return name.replace(name, short_to_full_dict[name])
```

Comment:
its very bad function, look here.
```suggestion
#a dictionary for storing all abbreviations and full forms
short_to_full_dict = {
    'р-н': 'район',
    'вул.': 'вулиця ',
    'бульв.': 'бульвар ',
    'просп.': 'проспект ',
    'пров.': 'провулок ',
    'пл.': 'площа ',
    'обл.': 'область',
    'области': 'області'
}
def change_to_full_name(name):
    name = name.lower()
    return short_to_full_dict.get(name, name)
```

Reply:
We also have to replace substrings

Reply:
doesn't this expression seem strange to you?
```
name.replace(name, short_to_full_dict[name])
```

Reply:
I am not an expert yet)